tensorflow/python/distribute/input_lib.py

```diff
@@ -35,6 +35,7 @@
 from tensorflow.python.distribute import input_ops
 from tensorflow.python.distribute import reduce_util
```

Comment:
This file still needs to address these main comments:

https://github.com/tensorflow/tensorflow/pull/38968#discussion_r486760778 (you've removed the placement on GPU but need to combine the code more with the other PER_WORKER path)
https://github.com/tensorflow/tensorflow/pull/38968/files#r486753816
https://github.com/tensorflow/tensorflow/pull/38968#discussion_r486759716

Once those are reviewed, i ll review it again.

Reply:
I tried to address the requests. Please have a look and let me know what you think.

Reply:
I don't think these are correctly addressed, i have left comments next to the relevant code.